config/opal_check_libnl.m4

```diff
@@ -0,0 +1,58 @@
+dnl -*- shell-script -*-
+dnl
+dnl Copyright (c) 2015      Research Organization for Information Science
+dnl                         and Technology (RIST). All rights reserved.
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+dnl OPAL_LIBNL_SANITY_INIT()
+dnl --------------------------------------------------------------------
+AC_DEFUN([OPAL_LIBNL_SANITY_INIT], [
+    opal_libnl_version=0
+    opal_libnl_libs=""
+    opal_libnl3_libs=""
+])
+
+dnl OPAL_LIBNL_SANITY_CHECK(lib, function, LIBS)
+dnl --------------------------------------------------------------------
+AC_DEFUN([OPAL_LIBNL_SANITY_CHECK], [
+    case $host in
+        *linux*)
+            AC_LANG_PUSH(C)
+            libnl_LIBS_save="$LIBS"
+            LIBS="$LIBS $3"
+            cat > conftest_c.$ac_ext << EOF
+extern void $2 (void);
+int main(int argc, char *argv[]) {
+    $2 ();
+    return 0;
+}
+EOF
+            OPAL_LOG_COMMAND([$CC -o conftest $CFLAGS $CPPFLAGS conftest_c.$ac_ext $LDFLAGS -l$1],
+                             [ldd_output=`ldd conftest`
+                              if echo $ldd_output | grep -q libnl.so; then
+                                  if test $opal_libnl_version -eq 3; then
+                                      AC_MSG_WARN([lib nl version conflict: $opal_libnl3_libs requires libnl-3 whereas $1 requires libnl])
+                                  else
+                                      opal_libnl_libs="$opal_libnl_libs $1"
+                                      opal_libnl_version=1
+                                  fi
+                              elif echo $ldd_output | grep -q libnl-3.so; then
```

Comment:
Shouldn't we test for *both* libnl and libnl3?  (for the case where a dependent library links against *both* libnl and libnl3)  It would be a very poor dependent library that did this, but we should catch that case, too, right?